client/signup/steps/import-url/index.jsx

```diff
@@ -2,165 +2,279 @@
+			// @TODO differentiate an unreachable site from one that's not compatible.
+			case 1000002:
+				return translate(
+					"We're not able to reach that site at this time, or it's not compatible with our URL importer."
+				);
+		}
+
+		return translate( 'There was an error with the importer, please try again.' );
 	};
 
-	handleInputBlur = () => {
-		this.setState( {
-			showValidation: true,
+	exitFlow = target => {
+		this.props.recordTracksEvent( 'calypso_signup_flow_exit', {
+			flow: this.props.flowName,
+			step: this.props.stepName,
+			target,
 		} );
+		page.show( target );
 	};
 
-	checkValidation( value ) {
-		const { translate } = this.props;
-		const message = isValidUrl( value )
-			? ''
-			: translate( 'Please enter the full URL of your site.' );
-
-		this.props.setValidationMessage( message );
-	}
+	recordSupportClicked = helpPage => {
+		this.props.recordTracksEvent( 'calypso_signup_support_clicked', {
+			flow: this.props.flowName,
+			step: this.props.stepName,
+			support_page: helpPage,
+		} );
+	};
 
 	renderContent = () => {
-		const { isInputDisabled, urlInputValidationMessage, urlInputValue, translate } = this.props;
-		const { showValidation } = this.state;
+		const { isLoading, urlInputValue, translate } = this.props;
+		const { showUrlMessage } = this.state;
+		const urlMessage = this.getUrlMessage();
+		const helpPage = 'https://en.support.wordpress.com/import/';
 
 		return (
-			<div className="import-url__wrapper">
-				<FormTextInputWithAction
-					placeholder={ translate( 'Please enter the full URL of your site.' ) }
-					action="Continue"
-					onAction={ this.handleAction }
-					label={ translate( 'URL' ) }
-					onChange={ this.handleInputChange }
-					defaultValue={ urlInputValue }
-					disabled={ isInputDisabled }
-					onBlur={ this.handleInputBlur }
-				/>
-				{ showValidation ? (
-					<FormInputValidation
-						text={ urlInputValidationMessage || translate( 'This URL is valid.' ) }
-						isError={ !! urlInputValidationMessage }
-					/>
-				) : (
-					<FormSettingExplanation>
-						{ translate( 'Please enter the full URL of your site.' ) }
-					</FormSettingExplanation>
-				) }
-			</div>
+			<Fragment>
+				<Card className="import-url__card" tagName="div">
+					<form className="import-url__form" onSubmit={ this.handleSubmit }>
+						<FormTextInput
+							className="import-url__url-input"
+							placeholder="https://example.wixsite.com/mysite"
+							disabled={ isLoading }
+							defaultValue={ urlInputValue }
+							onChange={ this.handleInputChange }
+							onBlur={ this.handleInputBlur }
+							inputRef={ this.handleInputRef }
+							isError={ !! urlMessage }
+						/>
+						<FormButton
+							className="import-url__submit-button"
+							disabled={ isLoading }
+							busy={ isLoading }
+							type="submit"
+						>
+							{ isLoading ? translate( 'Checking' ) + '...' : translate( 'Continue' ) }
```

Comment:
This could have unexpected results in RTL languages:
`translate( 'Checking' ) + '...'`

Is there any reason you didn't just include an ellipsis in the string to translate?

Also, probably better to use an ellipses character `…` instead of three dots.

Reply:
I started with `&hellip;` but struggled with it being escaped by `translate`. Got it figured out now.